src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1409,10 +1421,20 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
                             ParseDirectiveBlock(directiveBuilder, descriptor, parseChildren: (childBuilder, startingBraceLocation) =>
                             {
                                 NextToken();
-                                Balance(childBuilder, BalancingModes.NoErrorOnFailure, SyntaxKind.LeftBrace, SyntaxKind.RightBrace, startingBraceLocation);
-                                SpanContext.ChunkGenerator = new StatementChunkGenerator();
+
                                 var existingEditHandler = SpanContext.EditHandler;
-                                SpanContext.EditHandler = new CodeBlockEditHandler(Language.TokenizeString);
+                                if (Context.FeatureFlags.AllowRazorInAllCodeBlocks)
+                                {
+                                    SpanContext.EditHandler = new CodeBlockEditHandler(Language.TokenizeString);
+                                    var block = new Block(descriptor.Directive, directiveStart);
+                                    ParseCodeBlock(childBuilder, block);
+                                }
+                                else
+                                {
+                                    Balance(childBuilder, BalancingModes.NoErrorOnFailure, SyntaxKind.LeftBrace, SyntaxKind.RightBrace, startingBraceLocation);
```

Comment:
Do we need an edit handler for the legacy case?

Reply:
Ahh, woops totally fumbled that. Yes we do 👍👍